pym/bob/cmds/build.py

```diff
@@ -948,15 +958,41 @@ def _cookPackageStep(self, packageStep, checkoutOnly, depth):
                         wasDownloaded = True
                     elif self.__forcedDownload:
                         raise BuildError("Downloading artifact failed")
-                elif oldWasDownloaded:
+                elif oldWasDownloadedOrShared:
                     self._info("   PACKAGE   skipped (already downloaded in {})".format(prettyPackagePath))
                     wasDownloaded = True
 
+            if packageStep.isShared() and self.__sharedFolder and not wasDownloaded:
+                if BobState().getResultHash(prettyPackagePath) is None:
+                    sharedpackagepath = os.path.join(self.__sharedFolder, asHexStr(packageBuildId))
+                    # link into dist if shared package available
+                    if self._checkSharedPackage(packageStep, asHexStr(packageBuildId)):
+                        print(colorize("     Link from shared location {}".format(sharedpackagepath), "32"))
+                        dirlist = os.listdir(sharedpackagepath)
+                        cwd = os.getcwd()
+                        os.chdir(prettyPackagePath)
+                        for item in dirlist:
```

Comment:
No need to change the working directory (just use `os.path.join()`). Also there are only two things that must be linked: `workspace` and `audit.json.gz`.